src/buildx/install.ts

```diff
@@ -32,28 +33,47 @@ import {Docker} from '../docker/docker';
 import {Git} from '../git';
 import {Util} from '../util';
 
-import {GitHubRelease} from '../types/github';
+import {GitHubRelease, OctokitRelease} from '../types/github';
 
 export interface InstallOpts {
   standalone?: boolean;
+  gitAuthToken?: string;
+  octokit?: InstanceType<typeof Octokit>;
 }
 
 export class Install {
   private readonly _standalone: boolean | undefined;
+  private readonly _gitAuthToken: string | undefined;
+  private readonly _octokit: InstanceType<typeof Octokit> | undefined;
 
   constructor(opts?: InstallOpts) {
     this._standalone = opts?.standalone;
+    this._gitAuthToken = opts?.gitAuthToken;
+    this._octokit = opts?.octokit;
   }
 
   /*
    * Download buildx binary from GitHub release
-   * @param version semver version or latest
+   * @param version semver version or latest.
+   * @param owner Repo owner, defaults to 'docker'.
+   * @param repo Repo name, defaults to 'buildx'.
    * @returns path to the buildx binary
    */
-  public async download(version: string): Promise<string> {
-    const release: GitHubRelease = await Install.getRelease(version);
-    core.debug(`Install.download release tag name: ${release.tag_name}`);
+  public async download(version: string, owner?: string, repo?: string): Promise<string> {
+    const realOwner = owner || 'docker';
+    const realRepo = repo || 'buildx';
 
+    const release = await this.getReleaseMetadata(version, realOwner, realRepo);
```

Comment:
We should not always rely on the API. We got many issues in the past doing this for GHES users: https://github.com/docker/setup-buildx-action/issues/194 and also the API being broken: https://github.com/docker/setup-buildx-action/issues/190. That's why we rely on the assets CDN instead.

So looking at my previous comment I think we should call `getReleaseMetadata` only if a tagged release URL is detected otherwise we rely on `getRelease` as before if semver (will download from vanilla repo docker/buildx).

Reply:
So with this I think we should rename `getRelease` to `getReleaseFromJSON` and `getReleaseMetadata` to `getReleaseFromAPI`. WDYT?

Reply:
sgtm, done